lightningnetwork/lnd
# Modified the listchanneltxn subcommand

## Change Description
start_height must be <= end_height, error otherwise
transactions are returned in direct order (oldest first)
an additional flag --reverse is added to change the order (--reverse true, recent first)
Modified docs by modifying comments in the listchaintxn description command 

## Steps to Test
Carry out a debug installation using this command:
`make && make install`

Test command: 
`./lncli-debug --no-macaroons listchaintxns --start_height 200 --end_height 7000| grep block_height`
Expected outcome: transactions are returned in the order, oldest first
![image](https://user-images.githubusercontent.com/59079323/223863027-b5a9bbbf-db94-4e69-b375-0dc01f4a33c8.png)

`./lncli-debug --no-macaroons listchaintxns --start_height 7000 --end_height 200| grep block_height`
Expected outcome: Error
![image](https://user-images.githubusercontent.com/59079323/223863337-2d1d1f8a-07af-4331-bc7b-3718b6bb841b.png)

`./lncli-debug --no-macaroons listchaintxns --start_height 200 --end_height 7000 --reverse| grep block_height`
Expected outcome: transactions are returned in the reverse order,  newest first
![image](https://user-images.githubusercontent.com/59079323/223863771-adc8f2ea-40f9-411c-b428-e2693bc343ed.png)

**Change after review**

Test command: 
`./lncli-debug --no-macaroons listchaintxns --start_height 200 --end_height 7000 --reverse| grep block_height`
Expected outcome: transactions are returned in the order, oldest first
![image](https://user-images.githubusercontent.com/59079323/223863027-b5a9bbbf-db94-4e69-b375-0dc01f4a33c8.png)

`./lncli-debug --no-macaroons listchaintxns --start_height 7000 --end_height 200| grep block_height`
Expected outcome: Error
![image](https://user-images.githubusercontent.com/59079323/223863337-2d1d1f8a-07af-4331-bc7b-3718b6bb841b.png)

`./lncli-debug --no-macaroons listchaintxns --start_height 200 --end_height 7000| grep block_height`
Expected outcome: transactions are returned in the order,  newest  first
![image](https://user-images.githubusercontent.com/59079323/223863771-adc8f2ea-40f9-411c-b428-e2693bc343ed.png)
